### ISSUE TITLE
Include custom LaTeX macros for output formats other than `latex`

### ISSUE THREAD

**Issue by allefeld** [jgm/pandoc#7816]
I'm not sure if this is already possible, if not it's a feature request.

I have a collection of custom LaTeX macros in a file `definitions.tex`, tailored to the math I often write (linear algebra, statistics) which does not have complete support through `amsmath`. When I generate a PDF via LaTeX, these macros are included in the preamble by a command `${ definitions.tex() }` in my `latex` template.

Pandoc makes a lot of LaTeX math also usable for other output formats like HTML or Word `docx`. Is there a way to use this file of macros also in this case?

**Comment by jgm** [comment]
Yes, pandoc can parse LaTeX macro definitions and the result will be reflected in math even in formats other than tex.

Example:
```
 % pandoc --mathml
\newcommand{\my}{\pi}

$\my^2$
^D
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>π</mi><mn>2</mn></msup><annotation encoding="application/x-tex">\pi^2</annotation></semantics></math></p>
```
(I assume here you're converting  from markdown.)

So you just need to make sure that your macro definitions get into the markdown source that is to be parsed.  Depending on the details, you might just be able to pass definitions.tex as the first file on the command line:

```
pandoc definitions.tex content.md -f markdown -t html -s -o content.html
```
Here you'll need `-f markdown` because otherwise pandoc will guess it's latex.

**Comment by allefeld** [comment]
Thanks!

I had to fiddle around a bit to make it work with different output formats. When supplying `definitions.tex` as first input file, and I produce beamer output, the macros are copied verbatim into an initial frame. So now for HTML / docx I use this approach, and for LaTeX-based formats I use `--include-in-header definitions.tex`.

**Comment by allefeld** [comment]
I ran into a problem with the first-input-file approach if using reveal.js, i.e.

```
pandoc definitions.tex content.md -f markdown -t revealjs -s -o content.html
```

Though the LaTeX macros are absorbed by Pandoc (there is no resulting content), they lead to the creation of an initial empty slide (after the title slide and before the regular slides).

Is there a way to mark the macros as "read but do not create output"?

A workaround is this Lua filter:
```lua
-- To prevent LaTeX macros (already processed by Pandoc)
-- from creating an empty slide in Reveal.js.
function RawBlock(el)
  if el.format == "tex" then
    return {}
  end
  return el
end
```

**Comment by jgm** [comment]
Put them in a metadata field.

**Comment by allefeld** [comment]
> Put them in a metadata field.

I don't know how I would do that.
Which metadata field can be used for this?
Or can I just make one up?
Can I then still read the macros from a file?

**Comment by jgm** [comment]
Just make one up.

They can be in a file, but it must have the form of a pandoc YAML metadata block with the tex definitions in a metadata field.

**Comment by allefeld** [comment]
I tried that, and in the following file `test.md` it works with `pandoc --mathjax --standalone test.md -o test.html`.

```
---
title: Test Math Definitions
math-definitions: |
  \newcommand{\ud}{\mathup{d}}
---

$$
\int x ~ \ud x
$$
```

However, I have a long list of standard definitions, and it is inconvenient to put them into the header of each file. I therefore tried to move them out into a defaults file. I.e. `test.md`:

```
---
title: Test Math Definitions
---

$$
\int x ~ \ud x
$$
```

`math-definitions.yaml`:

```
math-definitions: |
  \newcommand{\ud}{\mathup{d}}
```

But I get an error:

```
$ pandoc --mathjax --standalone --defaults=math-definitions.yaml test.md -o test.html
Aeson exception:
Error in $: Unknown option "math-definitions"
```

Since that's probably because defaults-file entries are more restricted, I tried instead:

```
pandoc --mathjax --standalone --metadata-file=math-definitions.yaml test.md -o test.html
```

That gives no error, but the macro `\ud` is not processed and highlighted in red by MathJax.

What would be the correct way to include YAML-header math definitions from a separate file?

**Comment by jgm** [comment]
They won't work in a defaults file, because (as documented) fields in a defaults file's `metadata` section are parsed as plain text, rather than markdown.

But you can simply create a separate file like `mathdefs.md`:

```
---
math-defs: |
  \newcommand whatever...
...

```

and put this file on the command line before your main file.
